uefi/region.go

```diff
@@ -17,13 +17,13 @@ const (
 // Region holds the base and limit of every type of region. Each region such as the bios region
 // should point back to it.
 type Region struct {
-	Base  uint16
-	Limit uint16
+	Base  uint16 // Index of first 4k block
+	Limit uint16 // Index of last block
 }
 
 // Valid checks to see if a region is valid
 func (r *Region) Valid() bool {
-	return r.Limit > 0
+	return r.Limit > 0 && r.Limit >= r.Base
```

Comment:
Nice. Is this from the spec? Or was it something you thought of? Either way this is good, I just wanted to know.

Reply:
No, just something I thought of.

Is this Valid check supposed comply with a specific definition in the spec? If it is, we might want to make that clear in the function comment.

Reply:
no clue. you can look it up and tell me =P

Reply:
it wasn't thought with the specs in mind. Most of the original work on this library is coming from a mix of specs, UEFITool and uefi-firmware-parser